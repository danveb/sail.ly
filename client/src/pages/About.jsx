import { aboutSailing } from "../constants/about"; 
import aboutSailTop from "../assets/about/about.jpg"; 
import "../styles/About.css"; 

export const About = () => {
    return (
        <div className="about">
            <div className="about__wrapper">
                <div className="about__main">
                    <div className="about__top">
                        <img src={aboutSailTop} alt="sailing yacht in ocean" />
                    </div>
                    <div className="about__bottom">
                        {aboutSailing.map((item) => (
                            <div key={item.id}>
                                <h2>{item.header}</h2>
                                <p>{item.text_intro}</p>
                                <p>{item.text_main}</p>
                            </div>
                        ))}
                    </div>
                </div>
            </div>
        </div>
    )
}
