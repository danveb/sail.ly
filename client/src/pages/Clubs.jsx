// import { ClubList } from "../components/ClubList";
import { clubsIntro } from "../constants/clubs";
import clubIntro from "../assets/clubs/club-intro.jpg"; 
import "../styles/Clubs.css"; 
import { ClubList } from "../components/ClubList";

export const Clubs = () => {
    return (
        <>
            <div className="clubs">
                <div className="clubs__wrapper">
                    <div className="clubs__main">
                        <div className="clubs__top">
                            <img src={clubIntro} alt="sailing yacht in ocean" />
                        </div>
                        <div className="clubs__bottom">
                            {clubsIntro.map((item) => (
                                <div key={item.id}>
                                    <h2>{item.header}</h2>
                                    <p>{item.text_intro}</p>
                                    <p>{item.text_main}</p>
                                </div>
                            ))}
                        </div>
                    </div>
                </div>
            </div>
            <ClubList />
        </>
    )
}