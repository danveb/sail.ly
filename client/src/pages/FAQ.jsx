import { FAQItem } from "../components/FAQItem";
import faq from "../assets/faq/faq.jpg"; 
import "../styles/FAQ.css"; 

export const FAQ = () => {
    return (
        <div className="faq">
            <div className="faq__wrapper">
                <div className="faq__main">
                    <div className="faq__top">
                        <img src={faq} alt="sailing yacht in ocean" />
                    </div>
                    <div className="faq__bottom">
                        <FAQItem />
                    </div>
                </div>
            </div>
        </div>
    )
}