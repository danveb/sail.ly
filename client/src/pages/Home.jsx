import chevronDown from "../assets/home/chevron-down.svg"; 
import "../styles/Home.css"; 

export const Home = () => {
    return (
        <div>
            <div className="home">
                <div className="home__top">
                    <div className="home__top--title">
                        <h1>#SS | 23</h1>
                        <h3>Dream in open waters</h3>
                        {/* LINK to clubs */}
                    </div>
                    <div className="home__top--bg"></div>
                    <img src={chevronDown} alt="chevron down arrow" />
                </div>
            </div>
        </div>
    )
}