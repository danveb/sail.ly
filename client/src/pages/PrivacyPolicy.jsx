import "../styles/PrivacyPolicy.css"; 

export const PrivacyPolicy = () => {
    return (
        <div className="privacy">
            <div className="privacy__wrapper">
                <div className="privacy__main">
                    <div className="privacy__top" data-testid="privacy__top">
                        <h1>Privacy Policy</h1>
                        <p>At SAIL.LY we take your privacy seriously. This Privacy Policy explains how we collect, use, and disclose your personal information when you use our web app.</p>
                    </div>
                    <div className="privacy__bottom">
                        <h2>Information We Collect</h2>
                        <p>When you use SAIL.LY we do not collect any of the following personal information:</p>
                        <ul>
                            <li>Contact Information: Name and email address</li>
                            <li>Usage Information: Information about how you use our web app, such as the pages you visit and the features you use</li>
                            <li>Device Information: Information about the device you use to access our web app, such as your IP address, browser type, and operating system</li>
                        </ul>
                        <h2>How We Use Your Information</h2>
                        <p>We do not use your personal information to provide and/or improve our web app. This app is built and designed for anyone that may want to learn more about climatic conditions pertaining to sailing.</p>
                        <h2>Data Retention</h2>
                        <p>We do not retain any personal information simply because it is not needed.</p>
                        <h2>Changes to This Privacy Policy</h2>
                        <p>We may update this Privacy Policy if needed. If we make any material changes we will post a notice on our web app prior to the effective date of the changes.</p>
                    </div>
                </div>
            </div>
        </div>
    )
}