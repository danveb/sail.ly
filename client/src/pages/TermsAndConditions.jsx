import "../styles/TermsAndConditions.css"; 

export const TermsAndConditions = () => {
    return (
        <div className="terms" data-testid="terms">
            <div className="terms__wrapper">
                <div className="terms__main">
                    <div className="terms__top">
                        <h1>Terms And Conditions</h1>
                        <p>Welcome to SAIL.LY! Please read these Terms and Conditions carefully before using our web app. By accessing or using SAIL.LY, you agree to be bound by these Terms and Conditions. If you do not agree to these Terms and Conditions, please do not use our web app.</p>
                    </div>
                    <div className="terms__bottom">
                        <h2>Use of SAIL.LY</h2>
                        <p>SAIL.LY is an application intended for educational purposes. It serves anyone who has little-to-no experience up to experienced sailors.</p>
                        <h2>Intellectual Property</h2>
                        <p>You agree that all materials, products, and services provided on this payment application are the property of SAIL.LY, its affiliates, directors, officers, employees, agents, suppliers, or licensors including all copyrights, trade secrets, trademarks, patents, and other intellectual property.</p>
                        <h2>Applicable Law</h2>
                        <p>By visiting this website, you agree that the laws of the United States of America, without regard to principles of conflict laws, will govern these terms and conditions, or any dispute of any sort that might come.</p>
                    </div>
                </div>
            </div>
        </div>
    )
}