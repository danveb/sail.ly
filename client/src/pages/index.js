export { default as About } from "./About";
export { default as Clubs } from "./Clubs";
export { default as FAQ } from "./FAQ";
export { default as Home } from "./Home";
export { default as PrivacyPolicy } from "./PrivacyPolicy";
export { default as TermsAndConditions } from "./TermsAndConditions"; 