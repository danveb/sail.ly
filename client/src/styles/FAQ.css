.faq {
    background-color: #fff; 
}

.faq__wrapper {
    max-width: 1200px; 
    height: 100%; 
    margin: 0 auto; 
}

.faq__main {
    display: flex; 
    flex-direction: column; 
    justify-content: center; 
    align-items: center; 
    max-width: 590px; 
    margin: auto; 
}

.faq__top {
    display: flex; 
}

.faq__top img {
    width: 100%; 
    height: 100%; 
    object-fit: contain; 
}

.faq__bottom {
    margin: 30px 0px; 
}

@media screen and (max-width: 591px) {
    .faq__bottom {
        margin: auto; 
        padding: 25px; 
    }
}