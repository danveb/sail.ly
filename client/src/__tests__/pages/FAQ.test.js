import { render, screen } from "@testing-library/react";
import { FAQ } from "../../pages/FAQ"; 

describe("Clubs Component", () => {
    // smoke test
    test("renders without crashing", () => {
        render(<FAQ />); 
    });

    // snapshot test 
    test("matches snapshot", () => {
        const { asFragment } = render(<FAQ />); 
        expect(asFragment()).toMatchSnapshot(); 
    });

    // getByAltText
    test("returns correct alt text", () => {
        render(<FAQ />); 
        const altText = screen.getByAltText("sailing yacht in ocean"); 
        expect(altText).toBeTruthy(); 
    });
}); 