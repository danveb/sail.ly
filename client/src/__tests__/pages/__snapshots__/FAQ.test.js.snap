// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clubs Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="faq"
  >
    <div
      class="faq__wrapper"
    >
      <div
        class="faq__main"
      >
        <div
          class="faq__top"
        >
          <img
            alt="sailing yacht in ocean"
            src="faq.jpg"
          />
        </div>
        <div
          class="faq__bottom"
        >
          <div
            class="faqItem__drawer"
          >
            <div
              class="faqItem__drawer--close"
            >
              <div
                class="faqItem__drawer__control"
              >
                <h2
                  class="faqItem__drawer__heading"
                >
                  What is SAIL.LY?
                </h2>
                <div
                  class="arrow__container"
                >
                  <img
                    alt="arrow down"
                    class="arrow"
                    src="chevron-down-right.svg"
                  />
                </div>
              </div>
              <p
                class="faqItem__drawer__subheading"
              >
                SAIL.LY is a web app built and designed to share the joy of sailing to beginner sailors as an educational toolkit.
              </p>
            </div>
            <div
              class="faqItem__drawer--close"
            >
              <div
                class="faqItem__drawer__control"
              >
                <h2
                  class="faqItem__drawer__heading"
                >
                  Why SAIL.LY?
                </h2>
                <div
                  class="arrow__container"
                >
                  <img
                    alt="arrow down"
                    class="arrow"
                    src="chevron-down-right.svg"
                  />
                </div>
              </div>
              <p
                class="faqItem__drawer__subheading"
              >
                SAIL.LY allows users to research sailing clubs in the US and get detailed weather information and recommendation for sailing.
              </p>
            </div>
            <div
              class="faqItem__drawer--close"
            >
              <div
                class="faqItem__drawer__control"
              >
                <h2
                  class="faqItem__drawer__heading"
                >
                  Who can sign up on SAIL.LY?
                </h2>
                <div
                  class="arrow__container"
                >
                  <img
                    alt="arrow down"
                    class="arrow"
                    src="chevron-down-right.svg"
                  />
                </div>
              </div>
              <p
                class="faqItem__drawer__subheading"
              >
                SAIL.LY is a tool for education purposes and does not require users to sign up or register to this application. Simply check our 'Clubs' page to tour different sailing clubs and get current weather information and sailing recommendation.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
